### ISSUE TITLE
Lexer: Ending a “for of” or “for in” loop with a function in which you provide an object, causes the next line to result in an error

### ISSUE THREAD

**Issue by hawkerboy7** [jashkenas/coffeescript#4564]
The following [code](http://coffeescript.org/#try:for%20key%20of%20test%20a%3A%205%0A%20add%5Bkey%5D%20%3D%20true) fails to compile:

```coffeescript
for key of test a: 5
 add[key] = true
```

`Error on line 2: unexpected [`

Adding parentheses [works](http://coffeescript.org/#try:for%20key%20of%20test(a%3A%205)%0A%20add%5Bkey%5D%20%3D%20true):
```coffeescript
for key of test(a: 5)
 add[key] = true
```
And the one-liner [works too](http://coffeescript.org/#try:add%5Bkey%5D%20%3D%20true%20for%20key%20of%20test%20a%3A%205):
```coffeescript
add[key] = true for key of test a: 5
```
I feel this is a bug or is this expected behaviour?

**Comment by GeoffreyBooth** [comment]
[Minimal case](http://coffeescript.org/v2/#try:for%20key%20of%20fn%20k%3A%20v%0A%20%20fn%20key%0A):
```coffee
for key of fn k: v
  foo()

###
[stdin]:2:6: error: unexpected (
  foo()
     ^
```
Interestingly, [this compiles](http://coffeescript.org/v2/#try:fn%20key%20for%20key%20of%20fn%20k%3A%20v%0A):
```coffee
fn key for key of fn k: v
```

**Comment by helixbass** [comment]
@hawkerboy7 this is fixed on `2` via #4570 

@GeoffreyBooth this can be closed?

**Comment by GeoffreyBooth** [comment]
Yes! Thanks 😄

Fixed via #4570.
